src/IlluminateMessageRepository/IlluminateUuidV4MessageRepository.php

```diff
@@ -115,4 +121,30 @@ private function yieldMessagesForResult(Collection $result): Generator
 
         return isset($message) ? $message->header(Header::AGGREGATE_ROOT_VERSION) ?: 0 : 0;
     }
+
+    public function paginate(PaginationCursor $cursor): Generator
+    {
+        if ( ! $cursor instanceof OffsetCursor) {
+            throw new LogicException(sprintf('Wrong cursor type used, expected %s, received %s', OffsetCursor::class, get_class($cursor)));
```

Comment:
Small nit-picky thing, wouldn't `InvalidArgumentException` be better fitting here? 🤔